ferry.go

```diff
@@ -356,19 +377,26 @@ func (f *Ferry) RunStandaloneDataCopy(tables []*schema.Table) error {
 	dataIterator.AddBatchListener(f.BatchWriter.WriteRowBatch)
 	f.logger.WithField("tables", tables).Info("starting standalone table copy")
 
-	dataIterator.Run()
+	dataIterator.Run(context.TODO())
 
 	return nil
 }
 
 // Call this method and perform the cutover after this method returns.
-func (f *Ferry) WaitUntilRowCopyIsComplete() {
-	<-f.rowCopyCompleteCh
+func (f *Ferry) WaitUntilRowCopyIsComplete(ctx context.Context) {
+	select {
+	case <-ctx.Done():
+	case <-f.rowCopyCompleteCh:
+	}
 }
 
-func (f *Ferry) WaitUntilBinlogStreamerCatchesUp() {
+func (f *Ferry) WaitUntilBinlogStreamerCatchesUp(ctx context.Context) {
 	for !f.BinlogStreamer.IsAlmostCaughtUp() {
-		time.Sleep(500 * time.Millisecond)
+		select {
+		case <-ctx.Done():
+			return
```

Comment:
This seems super dangerous. We should at the very least return an `error` here, otherwise the caller would assume that the binlog streamer has caught up and they can proceed to cutover 💩